test/integration/test-amp-ad-3p.js

```diff
@@ -147,6 +163,18 @@ function createIframeWithApis(fixture) {
     return poll('wait for new IO entry', () => {
       return lastIO != null;
     });
+  }).then(() => {
+    let resolver;
```

Comment:
This can be simplified a lot:

```javascript
// ...
}).then(() => new Promise((resolve, reject) => {
  iframe.contentWindow.context.getHtml('a', ['href'], content => {
    if (content == '<a href="http://test.com/test">Test link</a>') {
      resolve();
    } else {
      reject(new Error('Invalid getHtml result: ' + content));
    }
  });
}));
```

I also made changes for a couple of nits here: 

1. No need for strict equality when comparing against a non-empty string.
2. message should say "invalid" rather than "incorrect".

Reply:
Good point.

Done.